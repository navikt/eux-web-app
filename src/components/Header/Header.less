.c-header {
  background-color: #99c2e8;
  display: flex;
  flex-direction: row;
  height: 4rem;
  justify-content: space-between;
  align-items: center;

  &__brand {
    align-items: center;
    display: flex;
    flex-direction: row;
  }

  &__skillelinje {
    border-left: 1px solid #02329c;
    display: flex;
    height: 30px;
  }

  &__tittel {
    color: #02329c;
    display: flex;
    font-size: 14pt;
    font-weight: bold;
    padding-left: 15px;
  }

  &__saksbehandler {

    align-items: flex-end;
    display: flex;
    flex-direction: row;
    margin-right: 1rem;
    align-items: center;

    .saksbehandler__navn {
      color: #02329c;
      font-weight: bold;
      display: flex;
      margin: auto 0;
      padding: 0.3em;
    }
  }
}

.highContrast .c-header {
  &__tittel {
    color: white;
  }
  &__saksbehandler {

    .c-header__highcontrast-link {
      color: white;
    }

    .saksbehandler__navn {
      color: white;
    }
  }
}
