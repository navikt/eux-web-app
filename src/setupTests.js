/* eslint-disable */

import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import each from 'jest-each';

global.fetch = require('jest-fetch-mock');

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.mount = mount;

global.each = each;
