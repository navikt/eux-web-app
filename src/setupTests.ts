import Enzyme, { mount, render, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { act } from 'react-dom/test-utils'
import { useDispatch, useSelector } from 'react-redux'
jest.mock('react-redux')

Enzyme.configure({ adapter: new Adapter() });

(global as any).shallow = shallow;
(global as any).render = render;
(global as any).mount = mount;
(global as any).act = act

// eslint-disable-next-line no-undef
HTMLCanvasElement.prototype.getContext = jest.fn()
window.scrollTo = jest.fn()
window.location.reload = jest.fn()
Object.defineProperty(window, 'getComputedStyle', {
  value: () => ({
    getPropertyValue: () => ('')
  })
})

jest.mock('i18next', () => {
  const use = jest.fn()
  const init = jest.fn()
  const loadLanguages = jest.fn()
  const result = {
    use: use,
    init: init,
    loadLanguages: loadLanguages
  }
  use.mockImplementation(() => result)
  return result
})

jest.mock('uuid', () => {
  v4: 'mock_uuid'
})

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}));

(useDispatch as jest.Mock).mockImplementation(() => jest.fn())

export const stageSelector = (defaultSelector: any, params: any) => {
  (useSelector as jest.Mock).mockImplementation(() => ({
    ...defaultSelector,
    ...params
  }))
}
