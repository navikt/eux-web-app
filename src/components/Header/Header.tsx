import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import NEESSILogo from 'resources/images/nEESSI';
import './Header.css';
import * as MPT from '../../proptypes/';

import { saksbehandlerSelectors } from '../../ducks/saksbehandler/';

const Header = (props: any) => {
  const { saksbehandler: { navn } } = props;
  console.log('werweewr')
  return (
    <header className="topplinje">
      <div className="topplinje__brand">
        <Link to="/">
          <NEESSILogo/>
        </Link>
        <div className="brand__skillelinje" />
        <div className="brand__tittel"><span>{process.env.REACT_APP_NAME}</span></div>
      </div>
      <div className="topplinje__saksbehandler">
        <div className="saksbehandler__navn">{navn}</div>
      </div>
    </header>
  );
};

Header.propTypes = {
  saksbehandler: MPT.Saksbehandler.isRequired,
};

const mapStateToProps = (state: any) => ({
  saksbehandler: saksbehandlerSelectors.SaksbehandlerSelector(state),
});

const mapDispatchToProps = () => ({
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
