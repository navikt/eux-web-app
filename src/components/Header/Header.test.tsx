import { render } from '@testing-library/react'
import React from 'react'
import { stageSelector } from 'setupTests'
import Header, { HeaderProps, HeaderSelector } from './Header'

jest.mock('react-router-dom', () => ({
  Link: () => (<div className='mock-link' />)
}))

describe('components/Header/Header', () => {
  let wrapper: any
  const initialMockProps: HeaderProps = {
    title: 'title',
    highContrast: false
  }

  const defaultSelector: HeaderSelector = {
    saksbehandler: undefined
  }

  beforeAll(() => {
    stageSelector(defaultSelector, {})
  })

  beforeEach(() => {
    wrapper = render(<Header {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('HTML structure', () => {
    expect(wrapper.exists('.c-header')).toBeTruthy()
  })
})
