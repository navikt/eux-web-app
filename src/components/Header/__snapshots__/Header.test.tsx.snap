// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header/Header Renders 1`] = `
<Header
  className="mock-classname"
>
  <header
    className="mock-classname c-header"
  >
    <div
      className="c-header__brand"
    >
      <Link
        className="ml-2 mr-2"
        onClick={[Function]}
        to="/"
      >
        <div
          className="mock-link"
        />
      </Link>
      <div
        className="c-header__skillelinje"
      />
      <div
        className="c-header__tittel"
      >
        <span>
          ui:app-name
        </span>
      </div>
    </div>
    <Undertittel>
      <n
        type="undertittel"
      >
        <h2
          className="typo-undertittel"
        >
          ui:app-title
        </h2>
      </n>
    </Undertittel>
    <div
      className="c-header__saksbehandler"
    >
      <Lenke
        className="c-header__highcontrast-link mr-3"
        href="#highContrast"
        onClick={[Function]}
      >
        <a
          className="lenke c-header__highcontrast-link mr-3"
          href="#highContrast"
          onClick={[Function]}
        >
          ui:label-highContrast
        </a>
      </Lenke>
    </div>
  </header>
</Header>
`;
